kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -105,36 +201,96 @@ extension [A, S](effect: A < S)
         effect.map(v => loop(v, 0))
     end repeatUntil
 
+    /** Performs this computation repeatedly with a retry policy in case of failures.
+      *
+      * @param policy
+      *   The retry policy to use
+      * @return
+      *   A computation that produces the result of this computation with Async and Abort[Throwable] effects
+      */
     def retry(policy: Retry.Policy)(using Flat[A], Frame): A < (S & Async & Abort[Throwable]) =
         Retry[Throwable](policy)(effect)
 
+    /** Performs this computation repeatedly with a limit in case of failures.
+      *
+      * @param limit
+      *   The limit to use
+      * @return
+      *   A computation that produces the result of this computation with Async and Abort[Throwable] effects
+      */
     def retry[S1](n: => Int < S1)(using Flat[A], Frame): A < (S & S1 & Async & Abort[Throwable]) =
         n.map(nPure => Retry[Throwable](Retry.Policy(_ => Duration.Zero, nPure))(effect))
 
+    /** Performs this computation repeatedly with a backoff policy and a limit in case of failures.
+      *
+      * @param backoff
+      *   The backoff policy to use
+      * @param limit
+      *   The limit to use
+      * @return
+      *   A computation that produces the result of this computation with Async and Abort[Throwable] effects
+      */
     def retry[S1](backoff: Int => Duration, n: => Int < S1)(using
         Flat[A],
         Frame
     ): A < (S & S1 & Async & Abort[Throwable]) =
         n.map(nPure => Retry[Throwable](Retry.Policy(backoff, nPure))(effect))
 
+    /** Performs this computation indefinitely.
+      *
+      * @return
+      *   A computation that never completes
+      * @note
+      *   This method is typically used for long-running processes or servers that should run continuously
+      */
     def forever(using Frame): Nothing < S =
         (effect *> effect.forever) *> (throw new IllegalStateException("infinite loop ended"))
 
+    /** Performs this computation when the given condition holds.
+      *
+      * @param condition
+      *   The condition to check
+      * @return
+      *   A computation that produces the result of this computation with Abort[Maybe.Empty] effect
+      */
     def when[S1](condition: => Boolean < S1)(using Frame): A < (S & S1 & Abort[Maybe.Empty]) =
         condition.map(c => if c then effect else Abort.fail(Maybe.Empty))
 
+    /** Performs this computation catching any Throwable in an Abort[Throwable] effect.
+      *
+      * @return
+      *   A computation that produces the result of this computation with Abort[Throwable] effect
+      */
     def explicitThrowable(using Flat[A], Frame): A < (S & Abort[Throwable]) =
```

Comment:
I prefer ZIO's naming `resurrect` - what do you think? 
It makes sense in the context of suppressing Throwables is an `orDie`. I'm open to other ideas.

Reply:
Good point. I'm also not very happy with the new success/fail/panic naming, it doesn't feel natural to me. Maybe we should just go with ZIO's naming for error handling. It's clear and it'd be easier for people coming from ZIO.

Reply:
@hearnadam I agree that `explicitThrowable` is no good. I wasn't aware of `resurrect`, but it would make sense to use that. We should also then use `orDie` instead of `implicitThrowable`. The alternative I had been mulling over was `throwAborts` (for `orDie`) and `abortThrows` (for `resurrect`). Happy to stick to ZIO convention, though.

@fwbrasil I don't see anything wrong with success/fail/panic except that success is a noun whereas fail is a verb. succeed/fail/panic or success/failure/panic would be marginally more graceful.

Reply:
@johnhungerford I am inclined to approve this CR to get the scaladocs published. Can you followup on some alignment with ZIO's naming?